sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -26,20 +26,50 @@ import org.apache.spark.sql.catalyst.expressions.SpecializedGetters
 import org.apache.spark.sql.catalyst.util.{ArrayData, DateTimeUtils, MapData}
 import org.apache.spark.sql.types._
 
+// `JackGenerator` can only be initialized with a `StructType` or a `MapType`.
+// Once it is initialized with `StructType`, it can be used to write out a struct or an array of
+// struct. Once it is initialized with `MapType`, it can be used to write out a map. An exception
+// will be thrown if trying to write out a struct if it is initialized with a `MapType`,
+// and vice verse.
```

Comment:
For this kind of comment, we use the style like:

    /**
     * Code comments...
     *
     */

Reply:
ok.  I'll modify it.